src/libstore/derivations.hh

```diff
@@ -33,6 +34,21 @@ typedef std::map<StorePath, StringSet> DerivationInputs;
 
 typedef std::map<string, string> StringPairs;
 
+// Bit:
+//  7: regular vs ca
+//  6: floating vs fixed hash if ca, regular always floating
+//  5: pure vs impure if ca, regular always pure
+//  _: Unassigned
+enum DerivationTypeAxis : uint8_t {
+    DtAxisCA = 0b10000000,
+    DtAxisFixed = 0b01000000,
+    DtAxisImpure = 0b00100000,
+};
+enum DerivationType : uint8_t {
+    DtRegular = 0b0000000,
+    DtCAFixed = 0b11100000,
+};
+
```

Comment:
Why do we need such an encoding? Why not just
```
struct DerivationType {
    bool isImpure = false;
    bool isFixedOutput = false;
    bool isCA = false;
    ...
};
```

Reply:
Well I wanted to emphasize that not all combinations are valid. But we could turn a derivation type into one of those rather than doing `&` tests.

Reply:
I'm not thrilled about this encoding either (well I don't mind having it internally, but it doesn't feel right to have it leaking outside).

Maybe just instead have a family of functions `bool is{CA,Fixed,Impure}(DerivationType derivationType)` so that the actual encoding stays local but we can still ensure that only a few combinations are valid

Reply:
Yes, that is just what I am thinking :)